numpy/core/tests/test_function_base.py

```diff
@@ -356,9 +356,13 @@ def test_retstep(self):
         for num in (0, 1):
             for ept in (False, True):
                 y = linspace(0, 1, num, endpoint=ept, retstep=True)
-                assert_(isinstance(y, tuple) and len(y) == 2 and
-                        len(y[0]) == num and isnan(y[1]),
-                        'num={0}, endpoint={1}'.format(num, ept))
+                msg = 'num={0}, endpoint={1}'.format(num, ept)
```

Comment:
The changes you've made here are consistent with what was already there, so this comment is really addressed at the old code, not your changes.  I don't see why the tests are checking only the length of `y[0]` and not the actual value.  I was going to suggest a few tweaks to your version, but I think it will be eaiser to just suggest a rewritten version of the entire test:

```
    def test_retstep(self):
        y = linspace(0, 1, 2, retstep=True)
        assert_(isinstance(y, tuple) and len(y) == 2)
        assert_array_equal(y[0], [0.0, 1.0])
        assert_equal(y[1], 1.0)

        for num in (0, 1):
            for ept in (False, True):
                y = linspace(0, 1, num, endpoint=ept, retstep=True)
                msg = 'num={0}, endpoint={1}'.format(num, ept)
                assert_(isinstance(y, tuple) and len(y) == 2, msg)
                assert_array_equal(y[0], [0.0][:num], msg)
                if num == 1 and not ept:
                    assert_equal(y[1], 1.0, msg)
                else:
                    assert_(isnan(y[1]), msg)
```

Reply:
I completely agree that the old tests are a bit odd, but I wanted to keep my changes minimal. I'd actually be more happy with the revised tests that you suggest.

Reply:
Feel free to change a bit more, also you can use `assert` statement instead of `assert_`. And I think the `msg` part is not very important anymore, pytest is always very verbose...

Reply:
Thanks for the suggestions. I rewrote the test and decided to also test the case `num=2` with `endpoint=False`, which was previously not tested.